libphal/phal_pdbg.C

```diff
@@ -300,24 +301,134 @@ bool isSbeVitalAttnActive(struct pdbg_target *proc)
 
 bool hasControlTransitionedToHost()
 {
-  //Read the scratch register to find if the control has moved to phyp (Host)
-  auto pibTarget = pdbg_target_from_path(nullptr, "/proc0/pib");
-
-  uint64_t l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
-  // HB changes the below core scratch reg as one of the last instructions that is run,
-  // so if that is zero then we're in phyp
-  uint64_t l_coreScratchRegAddr = 0x4602F489;
-
-  // Is there any error in reading the scom address, consider control is in hostboot
-  if (pib_read(pibTarget, l_coreScratchRegAddr, &l_coreScratchRegData))
-  {
-    // If unable to read the register, by default consider the control is in hostboot
-    log(level::ERROR, "scom read error: 0x%X", l_coreScratchRegAddr );
-    return false;
-  }
-
-  // If the register reads zero, return control moved to host.
-  return (l_coreScratchRegData == 0);
+	// Read the scratch register to find if the control has moved to phyp
+	// (Host)
+	auto pibTarget = pdbg_target_from_path(nullptr, "/proc0/pib");
+
+	uint64_t l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
+	// HB changes the below core scratch reg as one of the last instructions
+	// that is run, so if that is zero then we're in phyp
+	uint64_t l_coreScratchRegAddr = 0x4602F489;
+
+	// Is there any error in reading the scom address, consider control is
+	// in hostboot
+	if (pib_read(pibTarget, l_coreScratchRegAddr, &l_coreScratchRegData)) {
+		// If unable to read the register, by default consider the
+		// control is in hostboot
+		log(level::ERROR, "scom read error: 0x%X",
+		    l_coreScratchRegAddr);
+		return false;
+	}
+
+	// If the register reads zero, return control moved to host.
+	return (l_coreScratchRegData == 0);
+}
+
+using LocationCode = std::string;
+using FRUType = uint8_t;
+constexpr uint8_t EXP_LOCATIN_CODE_MIN_LENGTH = 17;
+constexpr uint8_t UNEXP_LOCATIN_CODE_MIN_LENGTH = 4;
+constexpr int continueTgtTraversal = 0;
+constexpr int reqAttrFound = 1;
+constexpr int reqAttrNotFound = 2;
+LocationCode g_locCode;
+
+std::optional<LocationCode> getUnexpandedLocCode(const std::string &locCode)
```

Comment:
If you are already using `using LocationCode = std::string` then make the function signature to tune to that to avoid unnecessary type confusion between the two. `std::optional<LocationCode> getUnexpandedLocCode(const LocationCode& locCode)`